main.cpp

```diff
@@ -543,6 +656,8 @@ struct CPU
     void overclockCpu(bool overclockedState = false);
     // launch an application
     bool launchApplication(std::string programName, double memorySize);
+    // new member function
+    CPU swapCpu(int threshold, int startingValue);
```

Comment:
a CPU instance will return another CPU?
that doesn't make much sense.
This is a member function, so the only way to call it is by using a CPU instance...
- [ ] revise